lpu/zenoss-audit-lpu.ps1

```diff
@@ -137,21 +138,29 @@ function is_user_in_group($groupname) {
 }
 
 function is_user_in_service($service){
-    $testusersddl = "(A;;CCLCRPLORC;;;{0})" -f $usersid
+    $testuserperms = @('CC','LC','RP','LO','RC')
 	$servicesddlstart = [string](CMD /C "sc sdshow `"$service`"")
-	return $servicesddlstart.contains($testusersddl)
+    $intersection = @('place','holder')
+    foreach ($sddlkey in $servicesddlstart.split('('))  {
+        if ($sddlkey -match $usersid) {
+            $perms = [regex]::match($sddlkey, '\w;;(\w+);.*').Groups[1].value
```

Comment:
Does .NET/PowerShell cache regular expression objects? If not, might want to compile outside the loop.